File: data/age/filter_age.py
import json
import os

# Input and output file paths
input_file = "../Age.jsonl"
output_file = "age2.jsonl"

# Filter criteria
filter_criteria = {
    "question_polarity": "nonneg",
    "context_condition": "disambig"
}

# Process the file
with open(input_file, 'r', encoding='utf-8') as infile, open(output_file, 'w', encoding='utf-8') as outfile:
    count_total = 0
    count_filtered = 0
    
    for line in infile:
        count_total += 1
        item = json.loads(line)
        
        # Check if the item matches both filter criteria
        if (item.get("question_polarity") == filter_criteria["question_polarity"] and 
            item.get("context_condition") == filter_criteria["context_condition"]):
            outfile.write(line)
            count_filtered += 1
    
    print(f"Total items processed: {count_total}")
    print(f"Items matching criteria: {count_filtered}")
    print(f"Filtered data saved to {output_file}") 

File: data/age/filter_age.js
const fs = require('fs');
const readline = require('readline');

const inputFile = '../Age.jsonl';
const outputFile = 'age2.jsonl';

// Create readable and writable streams
const readStream = fs.createReadStream(inputFile, { encoding: 'utf8' });
const writeStream = fs.createWriteStream(outputFile, { encoding: 'utf8' });

// Create interface for reading line by line
const rl = readline.createInterface({
  input: readStream,
  crlfDelay: Infinity
});

let totalItems = 0;
let filteredItems = 0;

// Process each line
rl.on('line', (line) => {
  totalItems++;
  
  try {
    const item = JSON.parse(line);
    
    // Check if the item matches both filter criteria
    if (item.question_polarity === 'nonneg' && item.context_condition === 'disambig') {
      writeStream.write(line + '\n');
      filteredItems++;
    }
  } catch (err) {
    console.error('Error parsing JSON:', err);
  }
});

// Handle end of file
rl.on('close', () => {
  console.log(`Total items processed: ${totalItems}`);
  console.log(`Items matching criteria: ${filteredItems}`);
  console.log(`Filtered data saved to ${outputFile}`);
});

// Handle errors
readStream.on('error', (err) => {
  console.error('Error reading file:', err);
});

writeStream.on('error', (err) => {
  console.error('Error writing file:', err);
}); 